src/parse_rare.ML

```diff
@@ -251,18 +255,15 @@ fun type_parser _ (("d"::"e"::"f"::"i"::"n"::"e"::"-"::"r"::"u"::"l"::"e"::"*"::
                   ::"r"::"u"::"l"::"e"::cs)) = (DEFINE_COND_RULE,cs) |
   type_parser l _ = raise_parser_error l "Could not read type!"
 
-(* In alf the rule type is determined by annotations and not by a keyword. However, there is some
-syntax sugar that allows to write declare-axiom for rules without any conditions. Since the type
-depends on whether a :rare-fixed-point or a :premises ((= F false)) term is found later in the
-rule we cannot determine it here yet. *)
+(* Eunoia declare-rule token *)
 
-datatype alf_rule_type = ALF_AXIOM | ALF_RULE
+datatype eunoia_declare_rule = EUNOIA_RULE
 
-fun alf_type_parser _ (("d"::"e"::"c"::"l"::"a"::"r"::"e"::"-"::"a"::"x"::"i"::"o"::"m":: cs)) =
-    (ALF_AXIOM,cs) |
-  alf_type_parser _ (("d"::"e"::"c"::"l"::"a"::"r"::"e"::"-"::"r"::"u"::"l"::"e":: cs)) =
-    (ALF_RULE,cs) |
-  alf_type_parser l _ = raise_parser_error l "Could not read type of alf rule!"
+fun eunoia_declare_rule_lexer 
```

Comment:
I'd prefer to call this `euonia_rule_type_parser `, since this is abstractly describing what it does and we might add more Eunoia support